run_project_tests.py

```diff
@@ -1247,11 +1247,16 @@ def _run_tests(all_tests: T.List[T.Tuple[str, T.List[TestDef], bool]],
                 testname += f' ({t.name})'
             should_fail = ''
             suite_args = []
+            fail_file = t.path / 'FAIL'
             if name.startswith('failing'):
                 should_fail = name.split('failing-')[1]
-            if name.startswith('warning'):
+            elif name.startswith('warning'):
```

Comment:
This is a bug fix, and should be split into a separate change.

Reply:
`name` can't start with both `failing` and `warning`, so this is only a cleanup.